src/core/lease.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the ""License"");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an ""AS IS"" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { MissingArgumentError, SoloError } from './errors.js'
+import { type V1Lease } from '@kubernetes/client-node'
+import { type K8 } from './k8.js'
+import { SECONDS } from './constants.js'
+import { LeaseHolder } from './lease_holder.js'
+import { LeaseAcquisitionError, LeaseRelinquishmentError } from './lease_errors.js'
+import { type LeaseRenewalService } from './lease_renewal.ts'
+
+export class Lease {
```

Comment:
As @instamenta mentioned, using the `_privateVar` and `get privateVar` naming convention